libraries/core_libs/consensus/src/dag/dag.cpp

```diff
@@ -414,6 +417,13 @@ void DagManager::addDagBlock(DagBlock const &blk, SharedTransactions &&trxs, boo
       trx_mgr_->saveTransactionsFromDagBlock(trxs);
       // Save the dag block
       db_->saveDagBlock(blk);
+
+      // TODO: This is an ugly temporary fix for testnet, a better solution is needed for dag block race condition
+      if (db_->getDagBlockPeriod(blk.getHash()) != nullptr) {
+        db_->removeDagBlock(blk.getHash());
+        LOG(log_er_) << "Block already in DB: " << blk.getHash();
+        return;
+      }
```

Comment:
I am not sure if you will fix it in a better way, or just leave it as this

Reply:
I will fix this but I will need another PR for this since the solution is not simple